docs/source/examples/pt_zne.md

```diff
@@ -0,0 +1,392 @@
+twirled_result = np.average(pt_vals)
+
+
+print(f"Error without twirling: {abs(ideal_value - noisy_value) :.3}")
+print(f"Error with twirling: {abs(ideal_value - twirled_result) :.3}")
+```
+
+## Combining Pauli Twirling with ZNE
+
+```{code-cell} ipython3
+from mitiq.zne import execute_with_zne
+
+executor=partial(execute, noise_level=NOISE_LEVEL)
+zne_pt_vals = []
+
+for i in twirled_circuits:
+    zne_pt_vals.append(execute_with_zne(i, executor))
+
+mitigated_result = np.average(zne_pt_vals)
+
+print(f"Error without twirling: {abs(ideal_value - noisy_value) :.3}")
+print(f"Error with ideal and twirling: {abs(ideal_value - twirled_result) :.3}")
+print(f"Error with ideal and ZNE + PT: {abs(ideal_value - mitigated_result) :.3}")
+
+```
+Accordingly, depending on the noise strength, a combination of PT and ZNE do not work that well compared to just PT or ZNE. Thus, it is important to understand when combining a noisy tailoring technique with an error mitigation technique provides a significant advantage. 
+
+```{code-cell} ipython3
+
+import matplotlib.pyplot as plt
+
+# Plot error vs noise strength
+noise_strength = np.linspace(0.0, 1.0, 50)
+error_no_zne_no_twirling = []
+error_with_twirling = []
+error_with_twirling_and_zne = []
+zne_vals = []
+ideal_values = []
+NUM_TWIRLED_VARIANTS = 30
+
+for strength in noise_strength:
+    ideal_value = execute(circuit, noise_level=0.0)
+    ideal_values.append(ideal_value)
+    noisy_value = execute(circuit, noise_level=strength)
+    id_noisy_diff = abs(ideal_value-noisy_value)
+    error_no_zne_no_twirling.append(id_noisy_diff)
+
+    twirled_circuits = generate_pauli_twirl_variants(circuit, num_circuits=NUM_TWIRLED_VARIANTS)
+    pt_vals = Executor(partial(execute, noise_level=strength)).evaluate(twirled_circuits)
+    twirled_result = np.average(pt_vals)
+    twirled_noisy_diff = abs(ideal_value-twirled_result)
+    error_with_twirling.append(twirled_noisy_diff)
+
+    executor=partial(execute, noise_level=strength)
+    zne_vals.append(abs(ideal_value - execute_with_zne(circuit, executor)))
+    
+    zne_pt_vals = []
+    for i in twirled_circuits:
+        zne_pt_vals.append(execute_with_zne(i, executor))
+    mitigated_twirled_result = np.average(zne_pt_vals)
+    error_with_twirling_and_zne.append(abs(ideal_value - mitigated_twirled_result))
+
+
+plt.plot(noise_strength, error_no_zne_no_twirling,"", label=r"|Ideal - Noisy|", color="#1f77b4")
+plt.plot(noise_strength, zne_vals, "", label=r"|Ideal - ZNE|", color="#bcbd22")
+plt.plot(noise_strength, error_with_twirling,"", label=r"|Ideal - Twirling|", color="#ff7f0e")
+plt.plot(noise_strength, error_with_twirling_and_zne, "", label=r"|Ideal - (ZNE + Twirling)|", color="#2ca02c")
+
+plt.xlabel(r"Noise strength, Coherent noise:$R_y(\frac{\pi}{2} \times \text{noise_strength})$")
+plt.ylabel("Expectation Values Difference")
+plt.title("Comparison of expectation values with ideal as a function of noise strength")
+plt.legend()
+plt.show()
```

Comment:
Do we understand why ZNE is so good when the noise strength = 0.5?

Reply:
I am not too sure for the circuit used in this example. 

If I use a smaller GHZ circuit of 2 qubits, the noise does not do anything as the final state vector before and after adding coherent noise is the same. Interestingly, this behavior is not preserved as the number of qubits in the initial ghz circuit is increased. Worth looking into!